client/src/components/TransactionItemAdder/TransactionItemAdder.tsx

```diff
@@ -0,0 +1,115 @@
+import React from "react";
+import {
+  Container,
+  TitleContainer,
+  Title,
+  Button,
+  TransactionForm,
+  InputContainer,
+  InputGroup,
+  Label,
+  Input,
+  FormButton,
+  ErrorContainer,
+} from "./TransactionItemAdder.styles";
+import { MdOutlineCancel } from "react-icons/md";
+import { useForm, SubmitHandler } from "react-hook-form";
+import * as yup from "yup";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { addItem } from "../../API/TransactionMethods";
+
+const schema = yup.object().shape({
+  title: yup.string().min(2).max(50).required("field is required"),
+  amount: yup
+    .number()
+    .typeError("must be a number")
+    .positive("must be positive")
+    .min(0)
+    .required("field is required"),
+});
+
+interface FormInputs {
+  title: string;
+  amount: number;
+  first_name?: string;
```

Comment:
Do you still need first_name and last_name?

Reply:
got rid of these now